src/main/java/io/anserini/eval/RelevanceJudgments.java

```diff
@@ -100,10 +116,80 @@ public Map<String, Integer> getDocMap(String qid) {
       return null;
     }
   }
+  private static String getCacheDir() {
+    File cacheDir = new File(CACHE_DIR);
+    if (!cacheDir.exists()) {
+      cacheDir.mkdir();
+    }
+    return cacheDir.getPath();
+  }
   
-  public static String getQrelsResource(Qrels qrels) throws IOException {
-    InputStream inputStream = RelevanceJudgments.class.getClassLoader().getResourceAsStream(qrels.path);
+  /**
+   * Method will return the qrel file as a string
+   * 
+   * @param qrelPath path to qrel file
+   * @return qrel file as a string
+   * @throws IOException
+   */
+  public static String getQrelsResource(Path qrelPath) throws IOException{
+    Path resultPath = qrelPath;
+    try{
+      resultPath = getQrelPath(qrelPath);
+    }catch (Exception e){
+      throw new IOException("Could not get qrel file from cloud or local file system");
+    }
+
+    InputStream inputStream = Files.newInputStream(resultPath);
     String raw = new String(inputStream.readAllBytes());
     return raw;
   }
+
+  /**
+   * Method will look for the absolute qrel path and return it as a Path object
+   * 
+   * @param qrelPath path to qrel file
+   * @return qrel path
+   * @throws IOException
+   */
+  private static Path getQrelPath(Path qrelPath) throws IOException{
+    if (!Qrels.contains(qrelPath)) {
+      // If the topic file is not in the list of known topics, we assume it is a local file.
+      Path tempPath = Path.of(getCacheDir() + "/" + qrelPath.getFileName().toString());
+      if (Files.exists(tempPath)) {
+        // if it is a unregistred topic in the Topics Enum, but it is in the cache, we use it
+        return tempPath;
+      }
+      return qrelPath;
+    }
+    
+    Path resultPath = getNewQrelAbsPath(qrelPath);
+    if (!Files.exists(resultPath)) {
+      resultPath = getQrelFromCloud(qrelPath);
+    }
+    return resultPath;
+  }
+
+  public static Path getNewQrelAbsPath(Path qrelPath){
+    return Paths.get(getCacheDir(), qrelPath.getFileName().toString());
+  }
+
+  /**
+   * Method will download the qrel file from the cloud and return the path to the file
+   * 
+   * @param qrelPath path to qrel file
+   * @return path to qrel file
+   * @throws IOException
+   */
+  public static Path getQrelFromCloud(Path qrelPath) throws IOException{
+    String qrelURL = CLOUD_PATH + "/" + qrelPath.getFileName().toString();
```

Comment:
use a system method to put together parts of path instead of just adding "/" - for better portability?

qrel -> qrels